src/cryptography/x509.py

```diff
@@ -669,6 +681,29 @@ class SubjectKeyIdentifier(object):
     def __init__(self, digest):
         self._digest = digest
 
+    @classmethod
+    def create_from_public_key(cls, public_key):
+        # This is a very slow way to do this.
+        serialized = public_key.public_bytes(
+            serialization.Encoding.DER,
+            serialization.PublicFormat.SubjectPublicKeyInfo
+        )
+        spki, remaining = decoder.decode(
+            serialized, asn1Spec=_SubjectPublicKeyInfo()
+        )
+        # the univ.BitString object is a tuple of bits. We need bytes and
+        # pyasn1 really doesn't want to give them to us. To get it we'll
+        # build an integer, hex it, then decode the hex.
+        bits = 0
+        for bit in spki.getComponentByName("subjectPublicKey"):
+            bits = bits << 1 | bit
+
+        # convert the integer to bytes
+        hex_string = '%x' % bits
+        n = len(hex_string)
+        data = binascii.unhexlify(hex_string.zfill(n + (n & 1)))
```

Comment:
We take the integer, convert to hex, zfill it to make sure it has an even number of characters, then unhexlify it to bytes. It's the reverse of `int.from_bytes(bytes, "big")`

```python
# python3
>>> import binascii
>>> value = 383842938479274659827346598273465298376452398476529038457
>>> hex_string = '%x' % value
>>> n = len(hex_string)
>>> data = binascii.unhexlify(hex_string.zfill(n + (n & 1)))
>>> data
b'\x0f\xa7\x81\xdc5\x1d\xd9\xa0%\x97\x964j\x01\xd8\xf1Z\xad\x93\xb1\xd4\x90\xf4y'
>>> int.from_bytes(data, "big")
383842938479274659827346598273465298376452398476529038457
>>> int.from_bytes(data, "big") == value
True
```